### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by juliogonzalez**
This PR:

- Performs a general cleaning (typos, deprecated stuff, clarifications -for example for -C/--check-)
- Adds a internal links between MD files (including to index), and external links, to make reader's life easier.
- Removes the Docker example (until we can provide en example which is easier to reproduce in real life), and adds one for Jenkins
- Moves all MD files to doc/ folder for easier management
- Cleans inline help (-h) to remove typos, clarifies some concepts and make GitBot more similar to usual UNIX commands and programs

For an easier view of the doc changes, I recommend visiting the branch: https://github.com/juliogonzalez/gitbot/tree/FIX/doc

### CODE REVIEW

**Comment by juliogonzalez** [comment]
Ok, it seems the tests waits for a command without arguments to return the help. However the behaviour is now changed so it only returns the actual error (providing hint on how to get help).

This is the standard behaviour for most unix commands:
```
$ cp
cp: missing file operand
Try 'cp --help' for more information.
```
Fixing the test.

**Comment by juliogonzalez** [comment]
My mistake.

It seems that the problem here is that if a mandatory option was missing, gitbot was return 0, while now returns 1 as standard unix commands:
```
$ cp
cp: missing file operand
Try 'cp --help' for more information.
$ echo $?
1
```

@MalloZup, I will try to fix it by myself, but I could use some help since I lack of experience with unit testing with rake :-)

File: tests/unit_tests/t_opt_parser.rb [hunk 1]
```diff
@@ -4,13 +4,13 @@
 
 # Test the option parser
 class GitbotOptionTest < Minitest::Test
-  def set_option(hash, s)
```
Comment (juliogonzalez) [inline tests/unit_tests/t_opt_parser.rb]: WARNING: I am not sure at all about this change, so waiting for @MalloZup input on how to fix this text correctly (as I understand it now checks the return code, but NOT the text).

**Comment by MalloZup** [comment]
@gitbot rerun check-option-test !!!

**Comment by MalloZup** [comment]
no changelog needed!

**Comment by MalloZup** [comment]
@gitbot rerun changelog_shouldpass !!!

**Comment by MalloZup** [comment]
@gitbot rerun check-option-test !!!

**Comment by MalloZup** [comment]
no changelog needed!

**Comment by MalloZup** [comment]
@gitbot rerun changelog_shouldpass !!!

**Comment by MalloZup** [comment]
@gitbot rerun check-option-test !!!

**Comment by MalloZup** [comment]
no changelog needed!

**Comment by MalloZup** [comment]
@gitbot rerun changelog_shouldpass !!!

**Comment by MalloZup** [comment]
@gitbot rerun check-option-test !!!

**Comment by MalloZup** [comment]
no changelog needed!

**Comment by MalloZup** [comment]
@gitbot rerun changelog_shouldpass !!!

**Comment by MalloZup** [comment]
@gitbot rerun check-option-test !!!

**Comment by MalloZup** [comment]
no changelog needed!

**Comment by MalloZup** [comment]
@gitbot rerun changelog_shouldpass !!!

**Comment by MalloZup** [comment]
@gitbot rerun check-option-test !!!

**Comment by MalloZup** [comment]
no changelog needed!

**Comment by MalloZup** [comment]
@gitbot rerun changelog_shouldpass !!!

File: doc/BASICS.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    -t, --test 'TEST.SH'             Command, or full path to script/binary to be used to run the test.
+    -f, --file '.py'                 pr_file type to run the test against: .py, .rb
+    -g, --git_dir 'GIT_LOCAL_DIR'    Specify a location where gitbot will clone the GitHub project. If the dir does not exists, gitbot will create one. For example: /tmp/
+
+Optional options:
+    -C, --check                      Check if there is any PR requiring a test, but do not run it.
+        --changelogtest              Check if the PR includes a changelog entry (Automatically sets --file ".changes").
+    -u, --url 'TARGET_URL'           Specify the URL to append to add to the GitHub review. Usually you will use an URL to the Jenkins build log.
+    -P                               '--PR 'NUMBER'
+                                     Specify the PR number instead of checking all of them. This will force gitbot to run the against a specific PR number,even if it is not needed (useful for using Jenkins with GitHub webhooks).
```
Comment (juliogonzalez) [inline doc/BASICS.md]: There was a typo here, as well as at lib/opt_parser.rb (last parenthesis was missing). It's now corrected.

**Comment by MalloZup** [comment]
yop looks good sofar, i will review it with calm tomorrow but seems really good! gracias! :dancer:

File: tests/unit_tests/t_opt_parser.rb (hunk 1 above)
Comment (MalloZup) [reply tests/unit_tests/t_opt_parser.rb]: yop correct

**Pull request merged**